### PR TITLE
Optionally exchange defined in consumer

### PR DESCRIPTION
**Pull request by mente**
This is an attempt to contribute changes made in #24 without actual command. This PR adds a functionality of consumer to define exchange. It allows to use consumer when exchange is not yet created. Tests updated.

Small comment updates included

### CODE REVIEW

**Comment by mente** [comment]
Also I thought to suggest `callback` functionality similar to old_sound bundle. But decided to make it on application level rather than library

File: src/AmqpBundle/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -185,4 +151,43 @@ protected function addConsumers(ArrayNodeDefinition $node)
                 ->end()
             ->end();
     }
+
+    private function exchangeOptions()
```
Comment (omansour) [inline src/AmqpBundle/DependencyInjection/Configuration.php]: please add some phpdoc ? did you run cocke ? https://github.com/M6Web/AmqpBundle/blob/master/CONTRIBUTING.md . Thanks !

**Comment by omansour** [comment]
@fabdsp WYT ?

File: src/AmqpBundle/DependencyInjection/Configuration.php (hunk 1 above)
Comment (mente) [inline src/AmqpBundle/DependencyInjection/Configuration.php]: Nope, I'll update it

File: src/AmqpBundle/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -80,31 +83,6 @@ protected function addProducers(ArrayNodeDefinition $node)
 
                                     // args
                                     ->arrayNode('arguments')
-                                    ->prototype('scalar')->end()
-                                    ->defaultValue(array())
-                                    ->normalizeKeys(false)
-                                    ->end()
-
-                                    // binding
-                                    ->arrayNode('routing_keys')
```
Comment (fabdsp) [inline src/AmqpBundle/DependencyInjection/Configuration.php]: Why have you removed the routing keys node here ?
this node is used here : https://github.com/M6Web/AmqpBundle/pull/25/files#diff-030150e6243abd1d512cc37c8c359dd8R105

But maybe i've missed something !

Comment (mente) [inline src/AmqpBundle/DependencyInjection/Configuration.php]: You've missed something! `Configuration::exchangeOptions()` :)

Comment (fabdsp) [inline src/AmqpBundle/DependencyInjection/Configuration.php]: tsssss sorry, i messed with the diff ! useless comment !
Thx !

**Comment by fabdsp** [comment]
:+1:

**Comment by lnahiro** [comment]
:+1:

**Comment by fabdsp** [comment]
v1.8.0

**Comment by mente** [comment]
:+1:

**Pull request merged**